### ISSUE TITLE
@sap/xssec depends on jsonwebtoken 8.5.1 which has several security issues

### ISSUE THREAD

**Issue by brodo** [SAP/cloud-sdk-js#3318]
**Describe the bug**
npm audit output:
```
# npm audit report

jsonwebtoken  <=8.5.1
Severity: high
jsonwebtoken unrestricted key type could lead to legacy keys usage  - https://github.com/advisories/GHSA-8cf7-32gw-wr33
jsonwebtoken has insecure input validation in jwt.verify function - https://github.com/advisories/GHSA-27h2-hvpr-p74q
jsonwebtoken's insecure implementation of key retrieval function could lead to Forgeable Public/Private Tokens from RSA to HMAC - https://github.com/advisories/GHSA-hjrf-2m68-5959
jsonwebtoken vulnerable to signature validation bypass due to insecure default algorithm in jwt.verify() - https://github.com/advisories/GHSA-qwph-4952-7xr6
No fix available
node_modules/jsonwebtoken
  @sap/xssec  >=3.0.0
  Depends on vulnerable versions of jsonwebtoken
  node_modules/@sap/xssec
    @sap-cloud-sdk/connectivity  *
    Depends on vulnerable versions of @sap/xssec
    node_modules/@sap-cloud-sdk/connectivity
      @sap-cloud-sdk/http-client  *
      Depends on vulnerable versions of @sap-cloud-sdk/connectivity
      node_modules/@sap-cloud-sdk/http-client
```

**To Reproduce**

1. run `npm i -s @sap-cloud-sdk/http-client @sap-cloud-sdk/connectivity`
2. run `npm audit`

**Expected behavior**
There should not be any security warnings when I run `npm audit`. Especially not with "Severity: high".

**Used Versions:**

- node version: 18.4.0
- npm version: 8.12.1
- SAP Cloud SDK version: 2.12.1

**Impact / Priority**

Affected development phase: Development

Impact: Blocked

**Comment by fwilhe** [comment]
Hi @brodo 

thanks for raising the issue. We are aware of the issue and we communicated this to the xsenv team. There is not yet an updated version, so we can't release an updated version now.

According to the xsenv team the way they use the jsonwebtoken library this is not exploitable, so the finding might be considered as a false positive in this case.

Still I agree that installing the package should not print the warning and we will update xsenv once the new version is available.

Best,
Florian

**Comment by jjtang1985** [comment]
Hi @brodo ,

Please also consider using [yarn resolutions](https://classic.yarnpkg.com/lang/en/docs/selective-version-resolutions/) or [npm overrides](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#overrides) (npm>=8.3.0) as a work around. Here is one stackoverflow [example](https://stackoverflow.com/questions/52416312/npm-equivalent-of-yarn-resolutions).

Best regards,
Junjie

**Comment by brodo** [comment]
@jjtang1985 Thanks for the tip!

**Comment by fwilhe** [comment]
Hi @brodo 

this issue was fixed in version 3.2.15 of @sap/xssec which you will get if you install the latest version of the Cloud SDK. Cf the pr here. https://github.com/SAP/cloud-sdk-js/pull/3332

Note that you might need to update your project's lockfile in order to get that version.

Best,
Florian
